### DISCUSSION TITLE
Foldable sidebar

### DISCUSSION THREAD

**Discussion by JGNi** [home-assistant/frontend#7658] [Other feature requests]
With the ever growing addons there keeps coming in HA, a foldable sidemenu is really necessary 
Sorted either by tags or drag-and-drop.

**Comment by frenck** [comment]
Re-ordering / organizing the side menu is already possible.

**Comment by JGNi** [comment]
> Re-ordering / organizing the side menu is already possible.

I must have missed that. Can you link to how it's done then?

**Comment by JGNi** [comment]
And not only re-ordering/organizing, but foldable....

**Comment by spacegaier** [comment]
> 
> 
> > Re-ordering / organizing the side menu is already possible.
> 
> I must have missed that. Can you link to how it's done then?

From your user profile => "Change the order and hide items from the sidebar" => "Edit"

You can also press and hold the header of the sidebar to activate edit mode.

**Comment by iantrich** [comment]
When you say "foldable" are you talking about nested sections under a label?

**Comment by JGNi** [comment]
> When you say "foldable" are you talking about nested sections under a label?

Yes. That's what I mean

/JGNi

**Comment by iantrich** [comment]
I don't see a need to add such complexity to our base. Can you better explain the use case you are trying to solve?

**Reply by JGNi** [reply]
When you're getting many menu items, you have to scroll a lot, to find the different menu items.
That was my idea behind suggesting it....
![image](https://user-images.githubusercontent.com/26222087/99154069-5ad46a00-26ad-11eb-8d43-999d2c92ef70.png)

**Reply by iantrich** [reply]
So instead of a scroll down you want a click to open? I don't see a big advantage there. Especially when you can re-arrange the sidebar to move your less used items to the bottom.

**Reply by KTibow** [reply]
In the meantime, you can create a lovelace dashboard in panel mode with a markdown card, put links to all of your panels, and then remove them from the sidebar.

**Reply by Mariusthvdb** [reply]
not being lazy here, but would you have a pointer to how to do this? an additional dashboard isn't the issue, the markdown in panel mode is. (preferably with the new `my:` tags/buttons?)

**Comment by Masterz69** [comment]
Have another idea: have 2-column sidebar.
One column dedicated for dashboards, second for admin staff - add-ons etc.

**Reply by spacegaier** [reply]
Again, same question as Ian posed above: What is the use case for that? Doesn't feel worth the effort to me. Will use up additional screen space, plus probably tricky to precisely tap on a touch screen.

**Reply by JGNi** [reply]
I have 20 menu items. I've seen those with a lot more. It would just be a big help to be able to organize them better. 
You know, I guess same idea as behind File Explorer in Windows, Linux, Mac? There must be a reason for them all, to make it the same way?
Better overview, better organizing....

**Reply by spacegaier** [reply]
As mentioned above you can hide the ones you don't need and change the order to have the ones you need most often on top. But I do in general understand where you were coming from. But I don't get this 2 column approach, hence my question in reply to @Masterz69.

**Reply by JGNi** [reply]
> ...and change the order to have the ones you need most often on top. 

I'll have to google how to do that. But I've also started looking into making a markdown card with links.

**Reply by spacegaier** [reply]
> 
> 
> > ...and change the order to have the ones you need most often on top.
> 
> I'll have to google how to do that. But I've also started looking into making a markdown card with links.

I explained that here in the discussion last November already 😉: https://github.com/home-assistant/frontend/discussions/7658#discussioncomment-127431

**Reply by Masterz69** [reply]
IMHO better usability = less movements & clicks.
It means UI with no scrolling up/down & left/right is better.

On mobile screen - dashoard is limited to ~8 views & and sidebar ~12 items.
On PC - more views before scrolling, but sidebar anyway is about the same size.

My case is simple - scrolling is not so good as no scrolling :)
